README.md

```diff
@@ -53,38 +48,147 @@ See the [Docker documentation](./docs/Docker.md) to get up and running
-4. Run the following command to set up the environment:
+  Q: Why "Upaya"?
+  A: "skill in means" https://en.wikipedia.org/wiki/Upaya
+
+4. Run the following command to set up your local environment:
 
   ```
   $ make setup
   ```
 
   This command copies sample configuration files, installs required gems
-  and sets up the database.
+  and sets up the database. Check out our Makefile commands learn more about what this command does: https://github.com/18F/identity-idp/blob/master/Makefile
 
-5. Run the app server with:
+5. Now that you have you have everything installed, you can run the following command to start your local server:
 
   ```
   $ make run
   ```
 
+  You should now be able to go to open up your favorite browser, go to `localhost:3000` and see your local development environment running.
+
+#### Running tests locally
+
+  Login.gov uses the following tools for our testing:
+
+  - [RSpec](https://relishapp.com/rspec/rspec-core/docs/command-line)
+  - [Guard](https://github.com/guard/guard-rspec)
+  - [Mocha documentation](https://mochajs.org/)
+
+  To run our full test suite locally, use the following command:
+
+  ```
+  $ make test
+  ```
+
+  Use the following command to run a subset of our test suite, excluding slower tests:
+
+  ```
+  $ make fast_test
+  ```
+
+  Check out our Makefile commands learn more about how you can customize this command to run specific tests using rspec: https://github.com/18F/identity-idp/blob/master/Makefile#L41
+
+##### Running smoke tests
+
+  The smoke tests are a series of RSpec tests designed to run against deployed environments. To run them against the local Rails server:
+
+  ```bash
+  ./bin/smoke_test --local
+  ```
+
+  To run the smoke tests against a deployed server, make sure you set up a `.env` file with the right configuration values, see [monitor_config.rb](spec/support/monitor/monitor_config.rb) for the full list of environment variables used. The script below will `source` that file and add the variables to the environment.
+
+  ```bash
+  MONITOR_ENV=INT ./bin/smoke_test --remote
+  ```
+
+#### Speeding up local development and testing
+
+  To automatically run the test that corresponds to the file you are editing,
+  run `bundle exec guard` with the env var `GUARD_RSPEC_CMD` set to your preferred
+  command for running `rspec`. For example, if you use [Zeus](https://github.com/burke/zeus),
+  you would set the env var to `zeus rspec`:
+  ```console
+  GUARD_RSPEC_CMD="zeus rspec" bundle exec guard
+  ```
+
+  If you don't specify the `GUARD_RSPEC_CMD` env var, it will default to
+  `bundle exec rspec`.
+
+  We also recommend setting up a shell alias for running this command, such as:
+  ```console
+  alias idpguard='GUARD_RSPEC_CMD="zeus rspec" bundle exec guard'
+  ```
+
+#### Viewing email messages
+
+  To view email messages, Mailcatcher must be running. You can check if it's
+  running by visiting http://localhost:1080/. To run Mailcatcher:
+
+  ```
+  $ mailcatcher
+  ```
```

Comment:
Mailcatcher is run as a background process through `make run` by default, can we say something along those lines? (Also encourages `make run` as a common development style)

```suggestion
  We include a tool called Mailcatcher to view email messages locally. When Mailcatcher is running, visit http://localhost:1080/ to see them.
  
  We spin up a mailcatcher process be default through `make run`, but if you want to run Mailcatcher as a standalone process, just run:

  ```
  $ mailcatcher
  ```
```

Reply:
(ok so the nested ``` blocks did not do well...please do not just accept this change as-is because it'll mess up the code block)

Reply:
@zachmargolis updated here 226713e

Reply:
thank you!